java/test/jmri/managers/AbstractManagerTestBase.java

```diff
@@ -109,11 +112,90 @@ public void vetoableChange(PropertyChangeEvent evt) throws PropertyVetoException
         Assert.assertEquals(base, l.getVetoableChangeListeners().length);
 
     }
+
     @Test
     public void testMakeSystemName() {
         String s = l.makeSystemName("1");
         Assert.assertTrue(s != null);
         Assert.assertTrue(! s.isEmpty());
     }
 
+    private Field getField(Class c, String fieldName) {
+        try {
+            return c.getDeclaredField(fieldName);
+        } catch (NoSuchFieldException ex) {
+            if (c.getSuperclass() != null)
+                return getField(c.getSuperclass(), fieldName);
+        }
+        
+        // Field not found
+        return null;
+    }
+    
+    @Test
+    public void testRegisterDuplicateSystemName() throws PropertyVetoException, NoSuchFieldException, NoSuchFieldException, IllegalArgumentException, IllegalAccessException {
+        if (l instanceof ProvidingManager)  {
+            ProvidingManager<E> m = (ProvidingManager<E>) l;
+            String s1 = l.makeSystemName("1");
+            String s2 = l.makeSystemName("2");
+            Assert.assertTrue(s1 != null);
+            Assert.assertTrue(! s1.isEmpty());
+            Assert.assertTrue(s2 != null);
+            Assert.assertTrue(! s2.isEmpty());
+
+            E e1;
+            E e2;
+
+            try {
+                e1 = m.provide(s1);
+                e2 = m.provide(s2);
+            } catch (IllegalArgumentException | NullPointerException | ArrayIndexOutOfBoundsException ex) {
+                // jmri.jmrix.openlcb.OlcbLightManagerTest gives a NullPointerException here.
+                // jmri.jmrix.openlcb.OlcbSensorManagerTest gives a ArrayIndexOutOfBoundsException here.
+                // Some other tests give an IllegalArgumentException here.
+
+                // If the test is unable to provide a named bean, abort this test.
+                JUnitAppender.clearBacklog(Level.WARN);
+                return;
```

Comment:
This should be `Assume.nullException(ex)`

Reply:
The method `Assume.nullException(ex)` doesn't exists.
But I tried `Assume.assumeNoException(ex);` instead but it doesn't work for what I'm trying to do here. For some tests, the call `m.provide(s1);` fails. And if it does, I cannot go further with the testing and I need to clear the log.

One example of a test that fails without this is `SerialSensorManagerTest`.